### DISCUSSION TITLE
References to preprocessing questions for semantic search

### DISCUSSION THREAD

**Discussion by qrdlgit** [milvus-io/milvus#27245] [Q&A and General discussion]
Looking for references to preprocessing questions for semantic search

For QA semantic search, the question is posed as a .. question. It seems like that would degrade similarity semantic search as knowledge bases are formatted as questions, are there any common approaches to solving this problem (preprocessing the question to a better format for similarity) that I can reference?

**Comment by yhmo** [comment]
We have a demonstration project for Retrieval Augmented Generation (RAG): https://github.com/zilliztech/akcio
The demonstration website is here: https://osschat.io/

In this project, we generate lots of questions by LLM, convert the questions to embeddings, and store embeddings into Milvus. When user commits his question, we find the most similar question in the db and get the answers.
https://github.com/zilliztech/akcio/blob/main/pics/osschat.png
